### PR TITLE
vapor 4

### PR DESCRIPTION
**Pull request by tanner0101**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,6 +1,5 @@
 .git
 .build
 DerivedData
-Package.resolved
```
Comment (nathanfallet) [inline .dockerignore:4]: This file (`Package.resolved`) should not be commited

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+DB_URL=postgres://vapor_username:vapor_password@localhost:5432/vapor_database
```
Comment (nathanfallet) [inline .env:1]: Not sure if `.env` should be committed
+ previous version was using sqlite not postgres

File: Sources/App/Models/Todo.swift [hunk 3]
```diff
@@ -1,26 +1,19 @@
...
 
-/// A single entry of a Todo list.
-final class Todo: SQLiteModel {
-    /// The unique identifier for this `Todo`.
-    var id: Int?
+final class Todo: Model, Content {
+    static let schema = "todos"
+    
+    @ID(key: .id)
+    var id: UUID?
```
Comment (nathanfallet) [inline Sources/App/Models/Todo.swift:8]: Why changing from `Int` id to `UUID`?

File: Sources/Run/main.swift [hunk 4]
```diff
@@ -1,3 +1,9 @@
 import App
+import Vapor
 
-try app(.detect()).run()
```
Comment (nathanfallet) [inline Sources/Run/main.swift:3]: Things done here shouldn't be in `app.swift`, `boot.swift` and `configure.swift` instead?

**Review by nathanfallet — COMMENTED** [review]
I'm not a professional with Vapor but some changes are not following what I've always seen in the documentation (eg: `app.swift` and some other files deleted, moving things to `main.swift`)

File: .dockerignore (hunk 1 above)
Comment (0xTim) [reply .dockerignore:4]: This should be added to Docker to ensure that the same version is deployed as tested with

File: Sources/App/Models/Todo.swift (hunk 3 above)
Comment (0xTim) [reply Sources/App/Models/Todo.swift:8]: Fluent's default is now UUID, it works better across all the supported databases

File: Sources/Run/main.swift (hunk 4 above)
Comment (0xTim) [reply Sources/Run/main.swift:3]: Bootstrapping the logging system and creating and running the app must be done in `main.swift`

**Comment by 0xTim** [comment]
Closing as this repo has been superseded by vapor/template - feel free to reopen on that repository if this PR still makes sense. Thanks for contributing!

**Pull request closed without merging**